### DISCUSSION TITLE
Provide generated sidebar entries to custom pages

### DISCUSSION THREAD

**Discussion by KianNH** [withastro/starlight#2146] [Feature Requests]
### What version of `starlight` are you using?

0.25.2

### What is your idea?

Provide generated sidebar entries, like how [`Astro.props.sidebar`](https://starlight.astro.build/reference/overrides/#sidebar) functions on overridden components, to custom pages (or more?).

### Why is this feature necessary?

When using the [`StarlightPage`](https://starlight.astro.build/guides/pages/#using-starlights-design-in-custom-pages) component, you can pass a `sidebar` prop with entries but have no way of grabbing the current entries to filter or append to.

It was an [use-case from the Discord](https://discord.com/channels/830184174198718474/1070481941863878697/1265349311160647720) to append entries to the sidebar, i.e blog posts on a `/src/pages/blog.astro` page.

Unfortunately, even if the [sidebar config](https://discord.com/channels/830184174198718474/1265357537096634378/1265380008923238583) is imported into the custom page then the shapes differ (i.e `{ label: string, link: string }` vs `{ type: "link", label: string, href: string, ... }`) and won't pass validation.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks again for flagging this @KianNH!

@HiDeoo what do you think about supporting the same config shape for the `sidebar` prop as in the main Starlight config? I would guess we can still autogenerate groups if needed? It might be a breaking change, but it could simplify use cases like this where people want to share a sidebar between their main Starlight config and their custom pages. And I don’t think it **prevents** any current use cases?

**Reply by HiDeoo** [reply]
> what do you think about supporting the same config shape for the sidebar prop as in the main Starlight config? I would guess we can still autogenerate groups if needed?

I think this is correct, and if I remember correctly and I'm not sure where, but I think we discussed some time ago this possibility as a potential improvement. This would indeed be a breaking change but mostly a shape change I think.

Noting that for the use case mentioned, this is already possible by not specifying a `sidebar` to `<StarlightPage/>` so it uses the default global sidebar, and appending entries in a `<Sidebar/>` override. Plugins wise, I think this would not have a lot of impact as it may still be easier for a similar behavior to also use the override approach vs grabbing the global sidebar config and passing it down to `<StarlightPage/>`.

When I get the time, I can try to play around with this and see if it works as we expect :thumbsup:

**Reply by delucis** [reply]
I also remember discussing this but went through the places I expected the discussion to be and couldn’t find it 😅 

It’s true you can do this with overrides, but I guess for some use cases with multiple `<StarlightPage>` it’s more comfortable to do manually and separate out the logic for each route instead of needing to add sidebar code for each route to the override.

**Comment by HiDeoo** [comment]
Starting with Starlight [`v0.26.0`](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.26.0), the `<StarlightPage />` component sidebar prop to accept an array of [SidebarItems](https://starlight.astro.build/reference/configuration/#sidebaritem) like the main Starlight `sidebar` configuration in `astro.config.mjs`. 

This change simplifies the definition of sidebar items in the `<StarlightPage />` component, allows for shared sidebar configuration between the global `sidebar` option and `<StarlightPage />` component, and also enables the usage of autogenerated sidebar groups with the `<StarlightPage />` component.
